small/small.c

```diff
@@ -162,27 +258,28 @@ smalloc(struct small_alloc *alloc, size_t size)
 {
 	small_collect_garbage(alloc);
 
-	struct factor_pool *upper_bound = factor_pool_search(alloc, size);
-	if (upper_bound == NULL) {
+	struct factor_pool *factor_pool = factor_pool_search(alloc, size);
+	if (unlikely(factor_pool == NULL)) {
```

Comment:
That's an optimization and is not relevant to current commit. Please move it to separate commit.